xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/page/events/PageNotificationEventListener.java

```diff
@@ -0,0 +1,170 @@
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.notifications.page.PageNotificationEvent;
+import org.xwiki.notifications.page.PageNotificationEventDescriptor;
+import org.xwiki.notifications.page.PageNotificationEventDescriptorContainer;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.ObservationManager;
+import org.xwiki.observation.event.AllEvent;
+import org.xwiki.observation.event.Event;
+import org.xwiki.security.authorization.AuthorExecutor;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Default implementation of {@link PageNotificationEventDescriptorContainer}.
+ *
+ * @version $Id$
+ * @since 9.4RC1
+ */
+@Component
+@Singleton
+@Named(PageNotificationEventListener.NAME)
+public class PageNotificationEventListener extends AbstractEventListener
+{
+    /**
+     * The listener name.
+     */
+    public static final String NAME = "Page Notification Event Listener";
+
+    @Inject
+    private ObservationManager observationManager;
+
+    @Inject
+    private PageNotificationEventDescriptorContainer pageNotificationEventDescriptorContainer;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private AuthorExecutor authorExecutor;
+
+    private class VelocityTemplateBooleanEvaluator implements Callable<Boolean>
+    {
+        private String content;
+
+        private Provider<XWikiContext> contextProvider;
+
+        /**
+         * Constructs a {@link VelocityTemplateBooleanEvaluator}.
+         *
+         * @param contextProvider a reference to the XWikiContext provider component
+         * @param content the content of the template that will be evaluated
+         */
+        VelocityTemplateBooleanEvaluator(Provider<XWikiContext> contextProvider, String content)
+        {
+            this.contextProvider = contextProvider;
+            this.content = content;
+        }
+
+        @Override
+        public Boolean call() throws Exception
+        {
+            return contextProvider.get().getWiki().evaluateVelocity(content, "page-notification").contains("true");
+        }
+    }
+
+    /**
+     * Constructs a {@link PageNotificationEventListener}.
+     */
+    public PageNotificationEventListener()
+    {
+        super(NAME, AllEvent.ALLEVENT);
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        boolean xObjectFound;
+
+        // Filter the event descriptors concerned by the event, then create the concerned events
+        for (PageNotificationEventDescriptor descriptor : pageNotificationEventDescriptorContainer.getDescriptorList())
+        {
+            // If the event is expected by our descriptor
+            if (descriptor.getEventTrigger().contains(event.getClass().getCanonicalName())) {
```

Comment:
Should also look at this : https://stackoverflow.com/questions/7495785/java-how-to-instantiate-a-class-from-string